packages/models.py

```diff
@@ -1,6 +1,12 @@
 class VersionedPackage:
-    def __init__(self, name: str, version: str, description: str, dependencies: list[tuple]):
+    def __init__(
+        self,
+        name: str,
+        version: str,
+        description: str,
+        dependencies: list["VersionedPackage"] | None = None
```

Comment:
Using "VersionedPackage" instead of "tuple" allows for more complex and recursive structures.

Reply:
Allowing None and setting an [] when None is provided ensures that `self.dependencies` is always a list, simplifying usage and avoiding potential errors from passing None.

Reply:
@alexbrebu this repository is for tech test code review exercises which we give ti candidates who want to work for Snyk. So the code deliberately contains areas that we hope a candidate who is applying to work may suggest to improve. 
If we post perfect clean code it will make it harder for us to use the candidates in tech tests.
If you are a candidate then we invite you to a session to show us your findings, posting on the PR makes your comments available to other candidates.
I hope you don't mind us deleting your comments for the above reasons cc @pablo-snyk